hub/core/chunk_engine.py

```diff
@@ -623,13 +631,32 @@ def validate_num_samples_is_synchronized(self):
         tensor_meta_length = self.tensor_meta.length
 
         # compare chunk ID encoder and tensor meta
-        chunk_id_num_samples = np.uint32(
-            self.chunk_id_encoder.num_samples if self.chunk_id_encoder_exists else 0
-        )
+
+        # update this if we change self.num_samples implementation later to use tensor meta length instead of chunk_id_encoder
+        chunk_id_num_samples = self.num_samples
+
         if tensor_meta_length != chunk_id_num_samples:
             commit_id = self.version_state["commit_id"]
             tkey = get_tensor_meta_key(self.key, commit_id)
             ikey = get_chunk_id_encoder_key(self.key, commit_id)
             raise CorruptedMetaError(
                 f"'{tkey}' and '{ikey}' have a record of different numbers of samples. Got {tensor_meta_length} and {chunk_id_num_samples} respectively."
             )
+
+    def _pop(self):
+        """Used only for Dataset.append"""
+        num_samples = self.num_samples
+        if num_samples == 0:
+            raise IndexError("pop from empty tensor")
+        self.commit_diff._pop()  # This will fail if the last sample was added in a previous commit
+        self._write_initialization()
+        chunk_ids, delete = self.chunk_id_encoder._pop()
+        if len(chunk_ids) > 1:  # Tiled sample, delete all chunks
+            del self.tile_encoder[num_samples - 1]
+        elif not delete:  # There are other samples in the last chunk
+            chunk_to_update = self.get_chunk(self.get_chunk_key_for_id(chunk_ids[0]))
+            chunk_to_update._pop_sample()
+        if delete:
+            for chunk_key in map(self.get_chunk_key_for_id, chunk_ids):
+                del self.cache[chunk_key]
+        self.tensor_meta.length -= 1
```

Comment:
is there a way to rollback min, max shape, auto inferred dtype, etc?

Reply:
Can roll back min shape and max shape if this was the first sample, but cant roll back dtype at the moment as we dont know if it was inferred or set by user.